File: backend/src/main/java/arbremojo/backend/service/PromoService.java
package arbremojo.backend.service;

import arbremojo.backend.model.Promo;
import arbremojo.backend.repository.PromoRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
public class PromoService extends EntityService<Promo, PromoRepository> {

    @Autowired
    public PromoService(PromoRepository entityRepository) {
        super(entityRepository);
    }

    @Transactional
    @Override
    public Integer deleteEntityById(Integer id) {
        return entityRepository.deleteEntityById(id);
    }
}

File: frontend/src/service/promo.service.ts
import {HttpClient} from "@angular/common/http";
import {EntityService} from "./entity.service";
import {Promo} from "../model/promo";
import {Injectable} from "@angular/core";

@Injectable({
  providedIn: 'root'
})
export class PromoService extends EntityService<Promo> {
  constructor(http: HttpClient) {
    super(http, "promo");
  }
}
